### PR TITLE
Use model facade

### LINKED ISSUE #59: [Model and packages] Connect model and controller
Task for #46 

# Connect model and controller

Change controller to use the model facade instead of individual parts of the model

### PR DESCRIPTION
**Pull request by filipkauffeldt**
Changed application and controller to use the model facade instead of directly accessing different parts of the model. This pull request will result in Entities not being drawn and friction not working, until these features are implemented in correct places.

Closes #59

### CODE REVIEW

**Review by Pontus-Wikstrom — APPROVED** [review]
LGTM

File: crocket/src/main/java/com/crocket/Model.java [hunk 1]
```diff
@@ -152,16 +166,53 @@ public void updateAim(int degrees) {
...
+        if (activePlayer == null) {
+            activePlayer = player;
+        }
+
+        entities.add(player.getBall());
+        movables.add(player.getBall());
     }
 
     public void clearPlayers() {
         players.clear();
```
Comment (Oskmey) [inline crocket/src/main/java/com/crocket/Model.java:204]: Might not be a problem but what would happen if we cleared and empty players list?

Comment (Oskmey) [reply crocket/src/main/java/com/crocket/Model.java:204]: Or does not have an instance

File: crocket/src/main/java/com/crocket/Model.java [hunk 2]
```diff
@@ -152,16 +166,53 @@ public void updateAim(int degrees) {
...
+        directionLine.decrementDegreeAngle();
+    }
+
     public void setPlayers(List<Player> players) {
         this.players = players;
+
+        activePlayer = players.get(0);
+
+        for (Player player : players) {
+            entities.add(player.getBall());
```
Comment (Oskmey) [inline crocket/src/main/java/com/crocket/Model.java:183]: This method closely resembles addPlayer. Can we consider incorporating it into this method?

File: crocket/src/main/java/com/crocket/Model.java (hunk 1 above)
Comment (filipkauffeldt) [reply crocket/src/main/java/com/crocket/Model.java:204]: I think it should work if the list is empty. Might be a problem if someone passes `null` in setPlayers(). I'll look into this

File: crocket/src/main/java/com/crocket/Model.java [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 public class Model implements IModel {
+    private static final double NO_MOVEMENT_THRESHOLD = 0.1;
+    
     private static Model instance = null;
     private List<Player> players;
     private Set<IMovable> movables;
```
[the remark this answers is not in the document]
Comment (filipkauffeldt) [reply crocket/src/main/java/com/crocket/Model.java:13]: It's used in update

File: crocket/src/main/java/com/crocket/Model.java (hunk 2 above)
Comment (filipkauffeldt) [reply crocket/src/main/java/com/crocket/Model.java]: Fixed

**Review by Oskmey — APPROVED** [review]
Nice, Nice

**Pull request merged**